### ISSUE TITLE
Badge example for mat-icon is not screen reader accessible

### ISSUE THREAD

**Issue by mfairchild365** [angular/components#14985]
#### What is the expected behavior?
The badge content associated with the `mat-icon` in the example should be available to screen reader users.

#### What is the current behavior?
The badge content associated with the `mat-icon` in the example is not available to screen reader users.

#### What are the steps to reproduce?
1. Using either of the following screen readers
   - NVDA 2018.4.1 + FireFox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [badge example page](https://material.angular.io/components/badge/examples)
3. Navigate to the "Icon with a badge" example
   - for NVDA, use the down arrow key to read the next item until you reach the icon
   - for iOS Voiceover, swipe right to read the next item until you reach the icon
4. Observe that the icon and badge content are not be announced

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
The documentation for badges states the following:

> When applying a badge to a <mat-icon>, it is important to know that the icon is marked as aria-hidden by default. If the combination of icon and badge communicates some meaningful information, that information should be surfaced in another way. See the guidance on indicator icons for more information.

This icon example appears to be an 'indicator icon' as described above, and should be marked up as such as to make the badge information available to screen reader users.

**Comment by crisbeto** [comment]
Seems like the issue was fixed, but the PR wasn't tagged properly so it didn't get closed automatically.
